src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,218 @@
+
+The conventional way to linearise the Jordan axiom is to equate coefficients (more formally, assume
+that the axiom holds in all field extensions). For simplicity we use brute force algebraic expansion
+and substitution instead.
+
+## References
+
+* [Cabrera García and Rodríguez Palacios, Non-associative normed algebras. Volume 1]
+  [cabreragarciarodriguezpalacios2014]
+* [Hanche-Olsen and Størmer, Jordan Operator Algebras][hancheolsenstormer1984]
+* [McCrimmon, A taste of Jordan algebras][mccrimmon2004]
+
+-/
+
+variables (A : Type*)
+
+/-- A (non-commutative) Jordan multiplication. -/
+class is_jordan [has_mul A] :=
+(lmul_comm_rmul : ∀ a b : A, (a * b) * a = a * (b * a))
+(lmul_lmul_comm_lmul : ∀ a b : A, (a * a) * (a * b) = a * ((a * a) * b))
+(lmul_lmul_comm_rmul : ∀ a b : A, (a * a) * (b * a) = ((a * a) * b) * a)
+(lmul_comm_rmul_rmul : ∀ a b : A, (a * b) * (a * a) = a * (b * (a * a)))
+(rmul_comm_rmul_rmul : ∀ a b : A, (b * a) * (a * a) = (b * (a * a)) * a)
+
+/-- A commutative Jordan multipication -/
+class is_comm_jordan [has_mul A]:=
+(mul_comm : ∀ a b : A, a * b = b * a)
+(jordan : ∀ a b : A, (a * b) * (a * a) = a * (b * (a *a)))
+
+/-- A (commutative) Jordan multiplication is also a Jordan multipication -/
+@[priority 100] -- see Note [lower instance priority]
+instance is_comm_jordan.to_is_jordan [has_mul A] [is_comm_jordan A] : is_jordan A :=
+{ lmul_comm_rmul := λ a b, by rw [is_comm_jordan.mul_comm, is_comm_jordan.mul_comm a b],
+  lmul_lmul_comm_lmul := λ a b, by rw [is_comm_jordan.mul_comm (a * a) (a * b),
+    is_comm_jordan.jordan, is_comm_jordan.mul_comm b (a * a)],
+  lmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm, ←is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm],
+  lmul_lmul_comm_rmul :=  λ a b, by rw [is_comm_jordan.mul_comm (a * a) (b * a),
+    is_comm_jordan.mul_comm b a, is_comm_jordan.jordan, is_comm_jordan.mul_comm,
+    is_comm_jordan.mul_comm b (a * a)],
+  rmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm b a, is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm], }
+
+
+/-- Semigroup multiplication satisfies the (non-commutative) Jordan axioms-/
+@[priority 100] -- see Note [lower instance priority]
+instance semigroup.is_jordan [semigroup A] : is_jordan A :=
+{ lmul_comm_rmul := λ a b, by rw mul_assoc,
+  lmul_lmul_comm_lmul := λ a b, by rw [mul_assoc, mul_assoc],
+  lmul_comm_rmul_rmul := λ a b, by rw [mul_assoc],
+  lmul_lmul_comm_rmul := λ a b, by rw [←mul_assoc],
+  rmul_comm_rmul_rmul := λ a b, by rw [← mul_assoc, ← mul_assoc], }
+
+@[priority 100] -- see Note [lower instance priority]
+instance comm_semigroup.is_comm_jordan [comm_semigroup A] : is_comm_jordan A :=
+{ mul_comm := mul_comm,
+  jordan := λ a b, mul_assoc _ _ _, }
+
+local notation `L` := add_monoid.End.mul_left
+local notation `R` := add_monoid.End.mul_right
+
+/-! The Jordan axioms can be expressed in terms of commuting multiplication operators -/
```

Comment:
@ocfnash, do you remember the instance name that's providing the `⁅x, y⁆ = x*y - y*x` operator here? I think it would be good to mention it in this comment.

Reply:
I think you mean `ring.has_bracket`, cf https://leanprover-community.github.io/mathlib_docs/algebra/lie/of_associative.html#ring.has_bracket

Reply:
Done.